feg/cloud/go/services/basic_acct/servicers/participants.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2021 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package servicers
+
+import (
+	"context"
+	"strings"
+
+	"github.com/golang/glog"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"magma/feg/cloud/go/feg"
+	"magma/feg/cloud/go/protos"
+	"magma/feg/cloud/go/serdes"
+	"magma/feg/cloud/go/services/feg/obsidian/models"
+	"magma/feg/cloud/go/services/feg_relay/gw_to_feg_relay"
+	"magma/orc8r/cloud/go/services/configurator"
+)
+
+const (
+	// Minimal length of PLMNID
+	MinPlmnIdLen = 5
+	MaxPlmnIdLen = 6
+)
+
+// RetrieveParticipants retrieves GW Network, FeG Network & GW logical ID from the caller's context
+func RetrieveParticipants(ctx context.Context, session *protos.AcctSession) (provider, consumer, gw string, err error) {
```

Comment:
is this function the reason why removed some fields from the protos?

By any chance, Is the bootstraper the one responsible on setting up this identity? Or is it sent in every single request?
http://github.com/magma/magma/blob/b8ac7839b660ac38a5a7949bc6bf031d5182f6d6/orc8r/gateway/go/services/bootstrapper/service/bootstrapper.go#L308-L308

Reply:
This one starts the process, but the cert SN and its GW identity association are created by the cloud bootstrapper service & this mapping only exists on the cloud, GW itself has no control over it.

Reply:
ooh so the way it is identified is through the cert SN👍